e2e/libs/page-init.ts

```diff
@@ -139,21 +174,80 @@ beforeEach(async () => {
 
   /** Emitted when an uncaught exception happens within the page. */
   page.on('pageerror', async (error: Error) => {
-    const title = await getPageTitle();
+    const title = await getPageTitle(page);
     try {
       logger.error(`PAGEERROR: "${title}"\n${error}`);
     } catch (err) {
       console.error(`❗ Exception when getting pageerror.\n${err}`);
     }
   });
-});
+};
 
-afterEach(async () => {
-  await page.setRequestInterception(false);
-  await jestPuppeteer.resetBrowser();
-});
+/**
+ *
+ * @param launchOpts: {@link LaunchOptions} New browser launch options.
+ */
+export const withBrowser = (launchOpts?: LaunchOptions) => async (
+  testFn: (browser: Browser) => Promise<void>
+): Promise<void> => {
+  console.log(`PUPPETEER_DEBUG: ${PUPPETEER_DEBUG}`);
+  console.log(`PUPPETEER_HEADLESS: ${PUPPETEER_HEADLESS}`);
+  console.log(`isDebug: ${isDebug}`);
 
-const getPageTitle = async () => {
+  const browser = await launch(launchOpts ? launchOpts : defaultLaunchOptions);
+  try {
+    console.log('withBrowser begin');
+    return await testFn(browser);
+  } catch (err) {
+    if (err instanceof Error) {
+      logger.error(err.message);
+    }
+    throw err;
+  } finally {
+    await browser.close().catch((err) => {
+      console.error(`Unable to close browser. Error message: ${err.message}`);
+    });
+    await new Promise((resolve) => setTimeout(resolve, 1000));
+  }
+};
+
+/**
+ * Launch new browser and incognito page. Opens Workbench Login page.
+ * @param launchOpts: {@link LaunchOptions} New browser launch options.
+ */
+export const withPage = (launchOpts?: LaunchOptions) => async (
+  testFn: (page: Page, browser: Browser) => Promise<void>
+): Promise<void> => {
+  let incognitoPage;
+  await withBrowser(launchOpts)(async (browser) => {
+    try {
+      console.log('withPage begin');
+      incognitoPage = await browser.createIncognitoBrowserContext().then((context) => context.newPage());
+      await setupPageBeforeEachTest(incognitoPage);
+      await new GoogleLoginPage(incognitoPage).load();
+      return await testFn(incognitoPage, browser);
+    } catch (err) {
+      if (err instanceof Error) {
+        logger.error(err.message);
+      }
+      await fs.ensureDir(failScreenshotDir);
```

Comment:
All of this stuff does feel a bit unrelated to a function called `withPage()`. Something like `rwTest()` might capture the intent better, but I'm also OK with the current naming. Alternatively , is there no other hook available for generic failure handling in puppeteer test cases?